### PR TITLE
[JENKINS-40344] Don't log warning for invalid crumb from anon

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-40344](https://issues.jenkins-ci.org/browse/JENKINS-40344).

Not worth the time to write a test for this IMO.

Untested, interactive test with PR build and two browser tabs (one with the ajax stuff, the other logging out) should do it.

### Proposed changelog entries

* Don't log warning when an anonymous user sends an invalid crumb, usually just an expired session. *(Or none because too minor)*

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests

### Desired reviewers

@rtyler @aheritier @oleg-nenashev

### CODE REVIEW

**Comment by daniel-beck** [comment]
Unrelated test failure.

@oleg-nenashev I think this is yours?

> Deprecated Agent Protocol Monitor should not be activated

**Review by oleg-nenashev — APPROVED** [review]

File: core/src/main/java/hudson/security/csrf/CrumbFilter.java [hunk 1]
```diff
@@ -68,18 +68,22 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
...
+
+            // JENKINS-40344: Don't spam the log just because a session is expired
+            Level level = Jenkins.getAuthentication() == Jenkins.ANONYMOUS ? Level.FINE : Level.WARNING;
+
             if (crumb != null) {
                 if (crumbIssuer.validateCrumb(httpRequest, crumbSalt, crumb)) {
                     valid = true;
                 } else {
-                    LOGGER.log(Level.WARNING, "Found invalid crumb {0}.  Will check remaining parameters for a valid one...", crumb);
+                    LOGGER.log(level, "Found invalid crumb {0}.  Will check remaining parameters for a valid one...", crumb);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/security/csrf/CrumbFilter.java]: Maybe it makes sense to also print authentication in the log, so the admin will know the user

**Review by oleg-nenashev — DISMISSED** [review]

**Review by aheritier — APPROVED** [review]
LGTM

**Comment by daniel-beck** [comment]
Still needs a manual test run.

**Comment by daniel-beck** [comment]
Still a false positive test failure.

**Comment by daniel-beck** [comment]
> Sep 28, 2017 1:37:32 PM WARNING hudson.security.csrf.CrumbFilter doFilter
> No valid crumb was included in request for /ajaxExecutors by anonymous. Returning 403.

**Comment by daniel-beck** [comment]
Now this actually works. `user` is the user name here, and I won't improve the log message.

----

An administrator enabled CSRF protection and an anon user didn't reload the page:

    Sep 28, 2017 1:58:28 PM FINE hudson.security.csrf.CrumbFilter
    No valid crumb was included in request for /jenkins/ajaxExecutors by anonymous. Returning 403.

A user logged in and still had a tab open from before they logged in (anonymous has Overall/Read):

    Sep 28, 2017 1:59:21 PM WARNING hudson.security.csrf.CrumbFilter doFilter
    Found invalid crumb 3fff1db905ef461b9ca15ffcd2d40997.  Will check remaining parameters for a valid one...
    Sep 28, 2017 1:59:21 PM WARNING hudson.security.csrf.CrumbFilter doFilter
    No valid crumb was included in request for /jenkins/ajaxBuildQueue by user. Returning 403.

A user logged out (or their session expired) and they have a tab open that sends the logged in CSRF crumb:

    Sep 28, 2017 1:59:47 PM FINE hudson.security.csrf.CrumbFilter
    Found invalid crumb ca440c8183756657930bd15406527f03.  Will check remaining parameters for a valid one...
    Sep 28, 2017 1:59:47 PM FINE hudson.security.csrf.CrumbFilter
    No valid crumb was included in request for /jenkins/ajaxBuildQueue by anonymous. Returning 403.

**Comment by daniel-beck** [comment]
Flaky unrelated test.

File: core/src/main/java/hudson/security/csrf/CrumbFilter.java (hunk 1 above)
Comment (oleg-nenashev) [reply core/src/main/java/hudson/security/csrf/CrumbFilter.java]: Only one message has been patched, but it's fine

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**